src/CalcManager/CEngine/calc.cpp

```diff
@@ -19,8 +26,8 @@ static constexpr long DEFAULT_RADIX = 10;
 
 static constexpr wchar_t DEFAULT_DEC_SEPARATOR = L'.';
 static constexpr wchar_t DEFAULT_GRP_SEPARATOR = L',';
-static constexpr wstring_view DEFAULT_GRP_STR = L"3;0";
-static constexpr wstring_view DEFAULT_NUMBER_STR = L"0";
+static const wstring_view DEFAULT_GRP_STR = L"3;0";
```

Comment:
why?

Reply:
Explained in 9bfb78e: Clang does not consider `char_traits<wchar_t>::length` a `constexpr` function:

```
/.../WindowsCalculator/src/CalcManager/CEngine/calc.cpp:29:31: error: constexpr variable 'DEFAULT_GRP_STR' must be initialized by a
      constant expression
static constexpr wstring_view DEFAULT_GRP_STR = L"3;0";
                              ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /.../WindowsCalculator/src/CalcManager/CEngine/calc.cpp:7:
In file included from /.../WindowsCalculator/src/CalcManager/pch_cross_platform.h:4:
In file included from /.../usr/include/c++/v1/sstream:174:
In file included from /.../usr/include/c++/v1/ostream:138:
In file included from /.../usr/include/c++/v1/ios:216:
In file included from /.../usr/include/c++/v1/__locale:15:
In file included from /.../usr/include/c++/v1/string:477:
/.../usr/include/c++/v1/string_view:236:31: note: non-constexpr function 'length'
      cannot be used in a constant expression
        : __data(__s), __size(_Traits::length(__s)) {}
                              ^
/.../WindowsCalculator/src/CalcManager/CEngine/calc.cpp:29:49: note: in call to 'basic_string_view(&L"3;0"[0])'
static constexpr wstring_view DEFAULT_GRP_STR = L"3;0";
                                                ^
In file included from /.../WindowsCalculator/src/CalcManager/CEngine/calc.cpp:7:
In file included from /.../WindowsCalculator/src/CalcManager/pch_cross_platform.h:4:
In file included from /.../usr/include/c++/v1/sstream:174:
In file included from /.../usr/include/c++/v1/ostream:138:
In file included from /.../usr/include/c++/v1/ios:216:
In file included from /.../usr/include/c++/v1/__locale:15:
In file included from /.../usr/include/c++/v1/string:477:
In file included from /.../usr/include/c++/v1/string_view:176:
/.../usr/include/c++/v1/__string:356:23: note: declared here
char_traits<wchar_t>::length(const char_type* __s) _NOEXCEPT
                      ^
3 warnings and 1 error generated.

** BUILD FAILED **
```

Reply:
https://github.com/llvm-mirror/libcxx/blob/master/include/__string#L130 (and for the wchar_t specialization: https://github.com/llvm-mirror/libcxx/blob/master/include/__string#L356 ) https://github.com/gcc-mirror/gcc/blob/master/libstdc%2B%2B-v3/include/bits/char_traits.h#L110 Both libcxx and gcc say that after c++14 char_traits::length is constexpr and is required for std::string_view for it to be so, so either you're doing this on mac with it's REALLY outdated libcxx, you're on an old version of clang, or you forgot to compile with --std=c++17.

Reply:
@chuggafan You are correct, I committed this before bumping the version to C++17. Fixed with the latest commit, thanks!